scripts/pylib/twister/twisterlib/coverage.py

```diff
@@ -223,6 +223,7 @@ def _generate(self, outdir, coveragelog):
         # We want to remove tests/* and tests/ztest/test/* but save tests/ztest
         cmd = ["gcovr", "-r", self.base_dir,
                "--gcov-ignore-parse-errors=negative_hits.warn_once_per_file",
+               "--merge-mode-functions=separate",
```

Comment:
@nordic-piks also proposed in this PR, could you look on it  ?
https://github.com/zephyrproject-rtos/zephyr/pull/66345/files#diff-d7590c92e90c3827d0cac169343ceaac1ba22223aec878f5ccef89ea3e035b1aR234

Reply:
@golowanow your proposal allows interesting possibilities.
Regarding merge-mode-functions, did you actually find it working, What gcovr are you using?
There is released gcovr 6.0 which does not include https://github.com/gcovr/gcovr/pull/844, thus passing this option will not help (I am still wonder why sometimes it helps).

Reply:
the `--merge-mode-function=separate` works to me with gcovr 6.0 to solve the same issue at `z_waitq_init()` compiled differently on ifdefs

Reply:
I'd suggest to add this option to other `gcovr` calls here as the issue happens at least for first run on tests and for the last one on merge and report

Reply:
Ok. Should I then extend my PR with similar changes like your - regarding merge-mode (which is ok for me)
or rather should I wait for your PR to be merged? (yours is bigger, the review may take a while)

Reply:
my PR might wait its review forever, so IMO lets fix this here

Reply:
Please let me know if that is ok, as I took a lot from your PR regarding printing gcvor calls.

Reply:
please just add the separate mode option in current code at gcovr run as you did, but not only at the first one, but other two, especially the last (third) execution.
the rest of refactoring in #66345 is not in the scope of this particular problem.

Reply:
I added to the first and the last gcovr call, the same like you did. Should it be enough?

Reply:
i mean now this PR has code changes not related to to the coverage merge conflicts, like additional logging and '-v' option etc. which are out of this scope.
My note is the gcovr merge option we need already for at least two gcovr invocations here.

Reply:
Indeed, I do not need "-v" - will remove it.
Additional logging is not explicit needed by will help a lot - as this was the only way that I noticed that the last gcovr call also needs that merge-mode option. Is you agree, I want to keep them.
Now I added merge-mode option for all gcovr calls.